app/src/main/kotlin/com/camerasecuritysystem/client/DashcamFragment.kt

```diff
@@ -239,7 +354,7 @@ class DashcamFragment : Fragment() {
                 activeRecording?.stop()
                 startRecording()
             } else {
-                throw RuntimeException("Unexpected state of recording")
+                throw UnexpectedStateException("Unexpected state of recording")
```

Comment:
Instead of a custom exception, I would use `assert(recording != null)`. This will crash when recording is null, both in Releasae and Debug. If you only want it to crash in Debug, you can also use our `cassert`.